solutions/gold/cses-1716.mdx

```diff
@@ -0,0 +1,96 @@
+---
+id: cses-1716
+source: CSES
+title: Distributing Apples
+author: Chuyang Wang
+---
+
+## Explanation
+
+To get a more illustrative idea of the given problem, consider $n + m - 1$ white
+balls in a row. Of those $n + m - 1$ white balls, $m - 1$ are chosen to be
+colored black as separators, which gives us exactly $m$ segments of (possibly
+zero) white balls. There are 
+
+$$
+\binom{n + m - 1}{m - 1}
+$$
+
+ways to choose the balls to color black. 
+
+The ways to color $m - 1$ balls black corresponds to the ways to distribute
+those $n$ apples to $m$ children, since the number of white balls in a segment
+corresponds to the number of apples given to a child.
+
+<Info title="Stars and bars"> 
+
+The given problem is equivalent to finding the number of solutions for the
+equation
+
+$$
+x_1 + x_2 + ... + x_n = m.
+$$
+
+The method used in this problem is called [Stars and
+bars](https://en.wikipedia.org/wiki/Stars_and_bars_(combinatorics)). 
+
+</Info>
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+
+const ll MOD = 1e9 + 7;
+const int MAXN = 2e6;
+
+vector<ll> fac(MAXN, 1);
+vector<ll> inv(MAXN, 1);
+
+ll qexp(ll base, ll exp) {
```

Comment:
can you use the guide's bin exp and send a link
consistency is always good

Reply:
which one do you prefer? cause there are multiple different versions here https://usaco.guide/gold/combo?lang=cpp

Reply:
i was thinking about the one [here](https://usaco.guide/gold/modular?lang=cpp)...